src/snowflake/snowpark/mock/_plan.py

```diff
@@ -314,6 +316,80 @@ def handle_order_by_clause(
     return result_df
 
 
+def validate_interval(interval: Optional[Union[Interval, int]]) -> bool:
+    """
+    Validates that the input value is valid for use as an interval.
+        None -> Unbounded interval either before or after the current row.
+        0 -> Use the current row and the bound.
+        Interval -> Use the Interval when calculating the bound.
+    """
+    if interval is None or interval == 0 or isinstance(interval, Interval):
+        return True
+    return False
+
+
+def negate_interval(interval: Interval) -> Interval:
+    """
+    Negates each value in an interval. Returns a new Interval object.
+    """
+    return Interval(
+        **{key.lower(): -value for key, value in interval.values_dict.items()}
+    )
+
+
+def compare_intervals(
+    lower: Optional[Union[Interval, int]], upper: Optional[Union[Interval, int]]
+) -> bool:
+    """
+    Returns true if lower >= upper.
+    """
+    if lower is None or upper is None:
+        return True
+
+    lower = lower or Interval()
+    upper = upper or Interval()
+
+    for part in [
+        "YEAR",
+        "QUARTER",
+        "MONTH",
+        "WEEK",
+        "DAY",
+        "HOUR",
+        "MINUTE",
+        "SECOND",
+        "MILLISECOND",
+        "MICROSECOND",
+        "NANOSECOND",
+    ]:
+        lower_part = lower.values_dict.get(part)
+        upper_part = upper.values_dict.get(part)
+
+        if lower_part or upper_part:
+            lower_part = lower_part or 0
+            upper_part = upper_part or 0
+            if (lower_part + upper_part) != 0:
+                return lower_part > upper_part
```

Comment:
depending on the desc, shouldn't this be?
```suggestion
                return lower_part >= upper_part
```

Reply:
You are correct. Fixed an added a test case that ensure this logic is correct.